File: ios/ofdpdfsigner_ios/ofdpdfsignerlib/util/Key.swift
//
//  Key.swift
//  ofdcreatorlib
//
//  Created by zizheng fan on 2024/5/31.
//

import Foundation



public class Key {
    
    public static let SM2_PRV_KEY : String = "SM2_Private"
    public static let SM2_PUB_KEY : String = "SM2_Public"
    public static let SM4 : String = "SM4"
    
    private var keyType : String = ""
    private var key : Data? = nil

    public init(keyType: String, key: Data) {
        self.keyType = keyType
        self.key = key
    }
  
    public init() {
    }
  
    public func getKey() -> Data? {
        return key
    }
  
    public func getKeyType() -> String? {
        return keyType
    }
  
    public func setKeyType(_ keyType: String) {
        self.keyType = keyType
    }
  
    public func setKey(_ key: Data) {
        self.key = key 
    }
}

File: java/ofdpdfsigner_jni/src/main/java/com/fzz/seal/entity/Key.java
package com.fzz.seal.entity;

public class Key {
    public static final String SM2_PRV_KEY = "SM2_Private";
    public static final String SM2_PUB_KEY = "SM2_Public";
    public static final String SM4 = "SM4";


    private String keyType;
    private byte[] key;

    public Key(String keyType, byte[] key) {
        this.keyType = keyType;
        this.key = key;
    }

    public Key() {

    }

    public byte[] getKey() {
        return key;
    }

    public String getKeyType() {
        return keyType;
    }

    public void setKeyType(String keyType) {
        this.keyType = keyType;
    }

    public void setKey(byte[] key) {
        this.key = new byte[key.length];
        System.arraycopy(key,0,this.key,0,key.length);
    }


}
